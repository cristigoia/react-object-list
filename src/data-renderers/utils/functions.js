/**
 * Gets the value of the cell from the row and keys provided
 * @param  {Object} row  Row supplied
 * @param  {Array|string} keys List of keys used to access the value
 * @return {(Object|string|number)}      Value from row, or row itself if no keys
 */
const getValueFromAccessor = (row, keys) => {
  if (!Array.isArray(keys)) keys = keys.split('.')
  let value = row
  const cloneKeys = [...keys]
  while (value && cloneKeys.length) {
    const thisKey = cloneKeys.shift()
    if (thisKey in value) {
      value = value[thisKey]
    }
  }
  return value
}

export {
  getValueFromAccessor,
}
