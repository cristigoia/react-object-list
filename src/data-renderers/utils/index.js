export {
  getValueFromAccessor,
  handleRowClick,
} from './functions'
