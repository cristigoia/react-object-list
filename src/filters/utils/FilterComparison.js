import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'

class FilterComparison extends React.Component {
  static propTypes = {
    options: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string,
      value: Select.propTypes.value,
    })),
    value: Select.propTypes.value,
    onChange: PropTypes.func.isRequired,
  }

  render() {
    const {options, value} = this.props
    return (
      <div className="objectlist-current-filter__filter-comparison">
        <Select
          options={options}
          value={value}
          onChange={newValue => this.props.onChange(newValue.value)}
          clearable={false}
        />
      </div>
    )
  }
}

export default FilterComparison
