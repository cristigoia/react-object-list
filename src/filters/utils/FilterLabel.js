import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component to render label for filters
 */
class FilterLabel extends React.Component {
  static propTypes = {
    /** Text to render in the label */
    label: PropTypes.string.isRequired,
  }
  render() {
    return (
      <label className="objectlist-current-filter__filter-name">{this.props.label}:</label>
    )
  }
}

export default FilterLabel
