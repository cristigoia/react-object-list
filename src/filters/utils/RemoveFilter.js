import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component to render button used to remove a filter
 */
class RemoveFilter extends React.Component {
  static propTypes = {
    /** Function to be called to remove filter */
    onClick: PropTypes.func.isRequired,
  }

  render() {
    return (
      <button className="objectlist-button objectlist-button--delete objectlist-button--icon objectlist-button--borderless" onClick={this.props.onClick} title="Remove Filter" />
    )
  }
}

export default RemoveFilter
