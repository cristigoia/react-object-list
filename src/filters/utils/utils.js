const possibleQueryStringLookupExpressions = ['exact', 'lt', 'gt', 'gte', 'lte']

export {
  possibleQueryStringLookupExpressions,
}
