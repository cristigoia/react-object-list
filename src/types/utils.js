function floatPrecision(floatValue, precision) {
  floatValue = parseFloat(floatValue)
  if (isNaN(floatValue)) { return parseFloat('0').toFixed(precision) } else {
    const power = Math.pow(10, precision)
    floatValue = (Math.round(floatValue * power) / power).toFixed(precision)
    return floatValue.toString()
  }
}

function commaNumber(numberString) {
  const parts = numberString.toString().split('.')
  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',')
  return parts.join('.')
}
/**
 * Converts a float to stringed currency representation
 * @param  {number} floatNumber Amount to convert
 * @return {string}             Dollar string
 */
export function currency(floatNumber) {
  return `${floatNumber < 0 ? '-' : ''}$${commaNumber(floatPrecision(Math.abs(floatNumber), 2))}`
}
