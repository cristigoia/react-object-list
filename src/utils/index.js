export * from './constants'
export * from './proptypes'
export * from './functions'
export loadingSpinner from './loading'
export ErrorMessage from './ErrorMessage'
