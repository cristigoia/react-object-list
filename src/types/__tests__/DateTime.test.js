import React from 'react'
import Moment from 'moment'
import { shallow } from 'enzyme'
import { snapshotTest } from 'utils/tests'
import { SHORTDATE_FORMAT, API_DATE_FORMAT } from '../../utils'
import { DateTime } from '../'

describe('DateTime', () => {
  it('parses date correctly', () => {
    const date = Moment('2017-01-01')
    const laterDate = Moment('2017-02-01')
    const testValues = [
      {},
      {empty: 'No content'},
      {value: date},
      {value: date, outputFormat: SHORTDATE_FORMAT},
      {value: '2018-02-07', outputFormat: SHORTDATE_FORMAT, dateOnly: true},
      {value: '2018-02-07 14:00:11', outputFormat: SHORTDATE_FORMAT, dateOnly: true},
      {value: [date, laterDate]},
      {value: {start: date, end: laterDate}},
      {value: {start: date, end: laterDate}, seperator: 'to'},
    ]
    testValues.forEach(testValue => {
      snapshotTest(<DateTime {...testValue} />)
    })
  })
  it('handles errors in utc', () => {
    const dateString = '2018-02-07'
    const instance = shallow(<DateTime dateOnly outputFormat="DD/MM/YY" />).instance()
    spyOn(Moment, 'utc').and.callFake(() => { throw new Error('BAD') })
    const result = instance.formatDateTime(dateString)
    expect(Moment.utc).toHaveBeenCalledWith(dateString, API_DATE_FORMAT)
    expect(result).toBe('07/02/18')
  })
})
