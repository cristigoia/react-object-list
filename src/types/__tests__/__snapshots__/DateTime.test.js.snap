// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTime parses date correctly 1`] = `
<div
  className="text-muted"
>
  -
</div>
`;

exports[`DateTime parses date correctly 2`] = `
<div
  className="text-muted"
>
  No content
</div>
`;

exports[`DateTime parses date correctly 3`] = `
<div
  className="text-muted"
>
  -
</div>
`;

exports[`DateTime parses date correctly 4`] = `
<div
  className="text-muted"
>
  -
</div>
`;

exports[`DateTime parses date correctly 5`] = `
<div>
  07/02/2018
</div>
`;

exports[`DateTime parses date correctly 6`] = `
<div>
  07/02/2018
</div>
`;

exports[`DateTime parses date correctly 7`] = `
<div>
  1st Jan 2017 12:00am - 1st Feb 2017 12:00am
</div>
`;

exports[`DateTime parses date correctly 8`] = `
<div>
  1st Jan 2017 12:00am - 1st Feb 2017 12:00am
</div>
`;

exports[`DateTime parses date correctly 9`] = `
<div>
  1st Jan 2017 12:00am to 1st Feb 2017 12:00am
</div>
`;
